File: go_src/9.go
func isPalindrome(x int) bool {
    if x < 0 || (x != 0 && x % 10 == 0) {
        return false
    }
    r := 0
    for x > r {
        r = r * 10 + x % 10
        x /= 10
    }
    return x == r || x == r / 10
}

File: java_src/9.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0 || (x % 10 == 0 && x != 0)) {
            return false;
        }

        int hold = x, reverse = 0;
        while (x > 0) {
            reverse = reverse * 10 + x % 10;
            x /= 10;
        }

        return hold == reverse;
    }
}
